servo/connectors/opsani_dev.py

```diff
@@ -214,6 +214,40 @@ async def check_kubernetes_service_type(self) -> None:
                 f"expected service type of ClusterIP or LoadBalancer but found {service.spec.type}"
             )
 
+    @servo.checks.require("Target container resource requests are within limits")
+    async def check_target_container_resources_within_limits(self) -> None:
+        # Load the Deployment
+        deployment = await servo.connectors.kubernetes.Deployment.read(
+            self.config.deployment,
+            self.config.namespace
+        )
+        # assert deployment, f"failed to read deployment '{self.config.deployment}' in namespace '{self.config.namespace}'"
+
+        # Find the target Container
+        target_container = None
+        for container in deployment.containers:
```

Comment:
There's a `deployment.find_container()` method available to cut this down.

But for some functional one-liner foo to do the same you can do `target = next(filter(lambda c: c.name == self.config.container, deployment.containers), None)`

`next` returns the next value in an iterator, filter returns a new iterator after applying a predicate, and `next` accepts a default argument returned upon no values available.